assembly/src/assembler/mod.rs

```diff
@@ -496,21 +526,24 @@ impl Assembler {
                     )
                     .with_span(proc_alias.span());
 
-                    let proc_mast_root = self.resolve_target(
-                        InvokeKind::ProcRef,
-                        &proc_alias.target().into(),
-                        &pctx,
-                        mast_forest_builder,
-                    )?;
+                    let proc_mast_root = {
+                        let node_id = self.resolve_target(
+                            InvokeKind::ProcRef,
+                            &proc_alias.target().into(),
+                            &pctx,
+                            mast_forest_builder,
+                        )?;
+                        mast_forest_builder.get_mast_node(node_id).unwrap().digest()
+                    };
 
                     // insert external node into the MAST forest for this procedure; if a procedure
                     // with the same MAST rood had been previously added to the builder, this will
                     // have no effect
-                    let proc_node_id = mast_forest_builder.ensure_external(proc_mast_root)?;
-                    let procedure = pctx.into_procedure(proc_mast_root, proc_node_id);
+                    let external_node_id = mast_forest_builder.ensure_external(proc_mast_root)?;
```

Comment:
Do we actually need this line here? It seems like we assume that the node already exists in the `mast_forest_builder` on line 536 above (would be good to add explanation there why `unwrap()` is OK). So, on line 536 we should already have the `mast_root` and `node_id`. Or am I missing something?

Reply:
Yeah that was an artifact of refactoring too many things at once - it was not necessary. Fixed it